apps/cowswap-frontend/src/modules/tokens/hooks/useEnoughBalance.ts

```diff
@@ -75,22 +75,37 @@ export interface EnoughBalanceParams extends Omit<UseEnoughBalanceParams, 'check
  * @param params Parameters to check balance and optionally the allowance
  * @returns true if the account has enough balance (and allowance if it applies)
  */
-export function hasEnoughBalanceAndAllowance(params: EnoughBalanceParams): boolean {
+export function hasEnoughBalanceAndAllowance(params: EnoughBalanceParams): boolean | undefined {
   const { account, amount, balances, nativeBalance, allowances } = params
+
+  if (!account || !amount) {
+    return undefined
+  }
+
   const isNativeCurrency = amount?.currency.isNative
   const token = amount?.currency.wrapped
   const tokenAddress = getAddress(token)
 
-  const balance = tokenAddress ? balances[tokenAddress]?.value : undefined
-  const allowance = (tokenAddress && allowances && allowances[tokenAddress]?.value) || undefined
```

Comment:
This is not an expensive operation, can remain in the hook's body IMO

Reply:
This wasn't an optimization. Look that i didn't memoize. 
It was just hard to read logic. I believe now the body of the IIFE is easier to read, and the result is just the single value we need

it was a nit on my side, just cause i thought those conditions were prone to errors

Reply:
In that case, can you use an external fn instead?
As I mentioned in the other comment, not found of this pattern.

Reply:
For me, this pattern works good in small functions like this one. Defining functions also have a bit of verbosity (define all the parameters, etc). Also some context switching for the reader. 

For me this function was not very complex, so it was ok, but I also find it ok to do the private functions here so decided to keep you happy 16f15cded907718e96b438595529f125473f7066